internal/repository/osbuildconfig/osbuildconfig.go

```diff
@@ -0,0 +1,57 @@
+package osbuildconfig
+
+import (
+	"context"
+	"github.com/project-flotta/osbuild-operator/api/v1alpha1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//go:generate mockgen -package=osbuildconfig -destination=mock_osbuildconfig.go . Repository
+type Repository interface {
+	Read(ctx context.Context, name string, namespace string) (*v1alpha1.OSBuildConfig, error)
```

Comment:
Same here - I don't think all of the methods are needed at this point. Please leave only used ones.